persistence/src/main/scala/hmda/persistence/processing/HmdaFileValidator.scala

```diff
@@ -233,21 +233,20 @@ class HmdaFileValidator(submissionId: SubmissionId) extends HmdaPersistentActor
       self ! LarValidationErrors(e.errors)
       self ! CompleteValidation(replyTo)
 
-    case CompleteValidation(replyTo) =>
+    case CompleteValidation(replyTo, originalSender) =>
       if (state.syntacticalErrors.isEmpty && state.validityErrors.isEmpty && state.qualityVerified && state.macroVerified) {
         log.debug(s"Validation completed for $submissionId")
-        replyTo ! ValidationCompleted(submissionId)
+        replyTo ! ValidationCompleted(originalSender)
       } else {
         log.debug(s"Validation completed for $submissionId, errors found")
-        replyTo ! ValidationCompletedWithErrors(submissionId)
+        replyTo ! ValidationCompletedWithErrors(originalSender)
       }
 
     case VerifyEdits(editType, v, replyTo) =>
       if (editType == Quality || editType == Macro) {
         persist(EditsVerified(editType, v)) { e =>
           updateState(e)
-          self ! CompleteValidation(replyTo)
-          sender() ! EditsVerified(editType, v)
+          self ! CompleteValidation(replyTo, Some(sender()))
```

Comment:
This could be a problem. The `sender()` might get created in a different thread than the original message processing due to the async nature of `persist`. To always be safe, the best thing to do in the first line after `VerifyEdits` is to "fix" the reference to the original sender, like so:

```scala
val client = sender()
```
And then replace `Some(sender())` with `Some(client)` (or whatever you want to call that `val`)